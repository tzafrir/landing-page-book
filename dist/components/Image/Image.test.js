'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _Image = require('./Image');

var _Image2 = _interopRequireDefault(_Image);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('Image', function () {
  var props = void 0;
  var mounted = void 0;
  var image = function image() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_Image2.default, props));
    }
    return mounted;
  };
  describe('Image', function () {
    beforeEach(function () {
      props = {
        alt: 'This is an image',
        src: 'logo.png'
      };
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_Image2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = image().find('img');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});