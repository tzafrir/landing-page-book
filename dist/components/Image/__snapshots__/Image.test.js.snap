// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Image should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="This is an image"
    border={false}
    className="img-fluid"
    imgFilter={null}
    rounded={false}
    shadow={false}
    src="logo.png"
    tooltip=""
    width={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <img
          alt="This is an image"
          className="img-fluid"
          onLoad={[Function]}
          src="logo.png"
          style={
            Object {
              "display": "none",
              "filter": null,
            }
          }
          title=""
          width={undefined}
        />,
        <svg
          alt="This is an image"
          className="img-fluid"
          viewBox="0 0 100 100"
          width={undefined}
        >
          <g>
            
            <rect
              fill="#CCC"
              height="100"
              width="100"
            />
          </g>
        </svg>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "This is an image",
          "className": "img-fluid",
          "onLoad": [Function],
          "src": "logo.png",
          "style": Object {
            "display": "none",
            "filter": null,
          },
          "title": "",
          "width": undefined,
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "This is an image",
          "children": <g>
            
            <rect
              fill="#CCC"
              height="100"
              width="100"
            />
          </g>,
          "className": "img-fluid",
          "viewBox": "0 0 100 100",
          "width": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <rect
                fill="#CCC"
                height="100"
                width="100"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "fill": "#CCC",
                "height": "100",
                "width": "100",
              },
              "ref": null,
              "rendered": null,
              "type": "rect",
            },
          ],
          "type": "g",
        },
        "type": "svg",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <img
            alt="This is an image"
            className="img-fluid"
            onLoad={[Function]}
            src="logo.png"
            style={
              Object {
                "display": "none",
                "filter": null,
              }
            }
            title=""
            width={undefined}
          />,
          <svg
            alt="This is an image"
            className="img-fluid"
            viewBox="0 0 100 100"
            width={undefined}
          >
            <g>
              
              <rect
                fill="#CCC"
                height="100"
                width="100"
              />
            </g>
          </svg>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "This is an image",
            "className": "img-fluid",
            "onLoad": [Function],
            "src": "logo.png",
            "style": Object {
              "display": "none",
              "filter": null,
            },
            "title": "",
            "width": undefined,
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "This is an image",
            "children": <g>
              
              <rect
                fill="#CCC"
                height="100"
                width="100"
              />
            </g>,
            "className": "img-fluid",
            "viewBox": "0 0 100 100",
            "width": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                <rect
                  fill="#CCC"
                  height="100"
                  width="100"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "fill": "#CCC",
                  "height": "100",
                  "width": "100",
                },
                "ref": null,
                "rendered": null,
                "type": "rect",
              },
            ],
            "type": "g",
          },
          "type": "svg",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
